src/wallet/rpcwallet.cpp

```diff
@@ -1133,6 +1133,9 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
     if (params.size() > 0)
         nMinDepth = params[0].get_int();
 
+    bool fFilterAddress = !fByAccounts && params.size() > 3;
+    const CBitcoinAddress filterAddress = fFilterAddress ?  CBitcoinAddress(params[3].get_str()) : CBitcoinAddress{};
```

Comment:
My personal preference is to keep the if statements rather than converting them to ternary conditionals, as I think that's clearer (and consistent with all the other RPCs). I'm not that concerned about having the parameters const.